store/src/java/com/zimbra/cs/service/mail/SendInviteReply.java

```diff
@@ -383,6 +385,9 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
                     apptFolderId = oldInv.isTodo() ? Mailbox.ID_FOLDER_TASKS : Mailbox.ID_FOLDER_CALENDAR;
                 MailSendQueue sendQueue = new MailSendQueue();
                 try {
+                    if (stat != null && IcalXmlStrMap.PARTSTAT_DECLINED.equals(stat) && !CalendarMailSender.VERB_DECLINE.equals(verb)) {
+                        calItem.setUpdatePrevFolders(true);
```

Comment:
As per discussion with @Prashantsurana, instead of setting flag, sendCalendarMessage() should be overloaded. Will update code and send PR again.